### ISSUE TITLE
Select not equal without spaces does not work

### ISSUE THREAD

**Issue by arve0** [mikefarah/yq#1711]
Hi and thanks for `yq` 🙌 I use it every day, it's great!

With multi-document, `yq` is unable to deselect a document when `!=` operator does not have space around it.

Example: 
```
❯ echo -e 'a: 1\n---\na: 2' | yq 'select(.a!=1)'
a: 1
---
a: 2
```

Expected output is `a: 2`.

Spaces around or before works as expected:
```
❯ echo -e 'a: 1\n---\na: 2' | yq 'select(.a != 1)'
a: 2

❯ echo -e 'a: 1\n---\na: 2' | yq 'select(.a !=1)'
a: 2
```

But not when space after, suggesting it is mixing .a and ! together:
```
❯ echo -e 'a: 1\n---\na: 2' | yq 'select(.a!= 1)'
a: 1
---
a: 2
```

Version of yq: 4.34.1
Operating system: mac
Installed via: homebrew

**Input Yaml**
data.yaml:
```yaml
a: 1
---
a: 2
```

**Command**
```
yq 'select(.a!=1)' data.yaml
```

**Actual behavior**

```yaml
a: 1
---
a: 2
```

**Expected behavior**

```yaml
a: 2
```

**Additional context**
Reported in #1377 also. Select with `==` without spaces works fine, for example `select(.a==1)`.

I have not looked at the yq code, but I can give the bug a try if I get a pointer for where to look 😊 Assuming this is considered a bug.

**Comment by mikefarah** [comment]
That is odd. Looking at the code it thinks that the exclamation mark is part of the property name. Will fix in the next version :)

**Comment by mikefarah** [comment]
Also - glad you like `yq` :)

**Comment by mikefarah** [comment]
Fixed in 4.35.2
